### PR TITLE
BaSP-MR-09-28: Subscription post

### PR DESCRIPTION
**Pull request by ghost**
subscription post

### CODE REVIEW

File: src/controllers/member.js [hunk 1]
```diff
@@ -71,7 +71,27 @@ const updateMember = (req, res) => {
...
+      return res.status(404).send('ID was not found');
+    }
+
+    await Member.findByIdAndDelete(id);
+
+    res.send('Member has been deleted');
+  } catch (error) {
+    res.status(500).send('Member could not be deleted');
+  }
+  return null;
```
Comment (NicoCagnina) [inline src/controllers/member.js:90]: we shouldn't return null here

File: src/controllers/member.js [hunk 2]
```diff
@@ -71,7 +71,27 @@ const updateMember = (req, res) => {
...
+
+    if (!memberExist) {
+      return res.status(404).send('ID was not found');
+    }
+
+    await Member.findByIdAndDelete(id);
+
+    res.send('Member has been deleted');
+  } catch (error) {
+    res.status(500).send('Member could not be deleted');
```
Comment (NicoCagnina) [inline src/controllers/member.js:88]: return this `return res.status(500).send('Member could not be deleted');`

File: src/controllers/member.js [hunk 3]
```diff
@@ -71,7 +71,27 @@ const updateMember = (req, res) => {
...
+
+    const memberExist = await Member.findById(id);
+
+    if (!memberExist) {
+      return res.status(404).send('ID was not found');
+    }
+
+    await Member.findByIdAndDelete(id);
+
+    res.send('Member has been deleted');
```
Comment (NicoCagnina) [inline src/controllers/member.js:86]: return this `return res.send('Member has been deleted');`

**Review by NicoCagnina — CHANGES_REQUESTED** [review]

File: src/controllers/subscription.js [hunk 4]
```diff
@@ -1,6 +1,25 @@
...
+  subscription.create({
+    classId,
+    members,
+    date,
+  })
+    .then((result) => res.status(201).json({
+      message: 'Subscription created',
+      result,
+    }))
+    .catch((error) => res.status(400).json({
```
Comment (NicoCagnina) [inline src/controllers/subscription.js:16]: this should be 500

**Review by NicoCagnina — CHANGES_REQUESTED** [review]

File: src/validations/subscription.js [hunk 5]
```diff
@@ -1,5 +1,26 @@
 const Joi = require('joi');
 
+const validateCreation = (req, res, next) => {
+  const SubscriptionValidation = Joi.object({
+    classId: Joi.string()
```
Comment (JuanMornik) [inline src/validations/subscription.js:5]: add a message in every validation

File: src/validations/subscription.js [hunk 6]
```diff
@@ -1,5 +1,26 @@
...
+      .hex()
+      .min(24),
+    members: Joi.string()
+      .required(),
+    date: Joi.date()
+      .max('now')
```
Comment (JuanMornik) [inline src/validations/subscription.js:11]: delete this line. The user may add a date after today

**Review by JuanCruzMoreira — CHANGES_REQUESTED** [review]

File: src/validations/subscription.js [hunk 7]
```diff
@@ -1,5 +1,25 @@
 const Joi = require('joi');
 
+const validateCreation = (req, res, next) => {
+  const SubscriptionValidation = Joi.object({
+    classId: Joi.string()
+      .hex()
+      .min(24),
+    members: Joi.string()
```
Comment (JuanMornik) [inline src/validations/subscription.js:8]: Member validations must be the same as in classId validations

**Review by JuanCruzMoreira — CHANGES_REQUESTED** [review]

**Review by JuanCruzMoreira — APPROVED** [review]

**Review by nicolascagnina — APPROVED** [review]

**Review by NicoCagnina — APPROVED** [review]

**Pull request merged**